multistep/commonsteps/step_http_server.go

```diff
@@ -111,13 +112,20 @@ func (s *StepHTTPServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Starting HTTP server on port %d", s.l.Port))
+	httpAddr = s.HTTPAddress
+	httpAddrIface, present := state.GetOk("http_ip")
+	if present && httpAddrIface.(string) != "" {
+		httpAddr = httpAddrIface.(string)
+	}
```

Comment:
To cast `string` only once
```suggestion
	httpAddrIface, present := state.Get("http_ip").(string)
	if present && httpAddrIface != "" {
		httpAddr = httpAddrIface
	}
```

Reply:
You can even simplify it to :

```suggestion
	httpAddrIface, _ := state.Get("http_ip").(string)
	if httpAddrIface != "" {
		httpAddr = httpAddrIface
	}
```